recipes/stlab/old/conanfile.py

```diff
@@ -0,0 +1,186 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get
+from conan.tools.scm import Version
+from conan.tools.build import check_min_cppstd
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.52.0"
+
+class Stlab(ConanFile):
+    name = 'stlab'
+    description = 'The Software Technology Lab libraries.'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/stlab/libraries'
+    license = 'BSL-1.0'
+    topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
+
+    settings = "arch", "os", "compiler", "build_type",
+
+    options = {
+        "boost_optional": [True, False],
+        "boost_variant": [True, False],
+        "coroutines": [True, False],
+        "task_system": ["portable", "libdispatch", "emscripten", "pnacl", "windows", "auto"],
+    }
+
+    default_options = {
+        "boost_optional": False,
+        "boost_variant": False,
+        "coroutines": False,
+        "task_system": "auto",
```

Comment:
Why auto? auto usually means that we don't understand how library works, and build is not predictable.

Reply:
It is for taking the library default argument. But this is for the old recipe. It is not my intention to change it. It is very unlikely someone download some of the old versions of this library.

Reply:
I mean that we should be able to predict what is the default argument of the library, don't you think? auto is almost always a bad option design, and it doesn't play well with package id.

Reply:
Oh I see it was already there. But there is also a new thread_system option with this auto value.

Reply:
Yeah, it was there, I just move it to another folder. 
About the `thread_system` option, you are right, I used the same idea as the `old` options. 
What do you suggest? To remove the `auto` value of the list?

Reply:
So, it is Ok to keep `auto` for the recipe in `old` folder and to remove `auto` in the new recipe (`all` folder)?

Reply:
Only if you can predict from settings the correct default option (and therefore set this value in config_options() if it's conditional).

Reply:
I will try to use the STLab (original code) default, if they have a fixed default. 
What if I can predict the default (like, the default depends in a not simple logic)? What can I do in this case?
And...
I don't understand your comment about `config_options()`...